messaging/src/test/java/org/axonframework/utils/AssertUtils.java

```diff
@@ -51,4 +58,40 @@ public static void assertWithin(int time, TimeUnit unit, Runnable assertion) {
             now = System.currentTimeMillis();
         } while (true);
     }
+
+    /**
+     * Assertion utility using {@link org.awaitility.Awaitility}, awaiting the successful completion of the given
+     * {@code future} for 500ms at poll intervals of 25ms.
+     *
+     * @param future The completable future to await with {@link org.awaitility.Awaitility}.
+     * @param <R>    The result of the given {@code future}.
+     * @return The result from the given {@code future}.
+     */
+    public static <R> R awaitSuccessfulCompletion(@Nonnull CompletableFuture<R> future) {
+        await().atMost(Duration.ofMillis(500))
+               .pollDelay(Duration.ofMillis(25))
+               .untilAsserted(() -> assertFalse(
+                       future.isCompletedExceptionally(),
```

Comment:
This is not correct though, right? This is always false while still running. 

I think we need to wait on `isDone`, right?

Reply:
Yes, but if the task takes longer than 25ms (the initial poll dealy), the flag would also be true (it didn't complete, so neither exceptionally). 

So I would expect something like:
```java
await().atMost(Duration.ofMillis(500))
               .pollDelay(Duration.ofMillis(25))
               .untilAsserted(() -> assertTrue(
                       future.isDone() && !future.isCompletedExceptionally(),
```

Reply:
Ah fair, you were voting for a combo, not a replacement. Sure, good catch actually!